x/incentive/keeper/mock_test.go

```diff
@@ -148,7 +149,7 @@ func (m *mockLeverageKeeper) GetCollateral(_ sdk.Context, addr sdk.AccAddress, d
 func (m *mockLeverageKeeper) DonateCollateral(ctx sdk.Context, addr sdk.AccAddress, uToken sdk.Coin) error {
 	newCollateral := m.GetCollateral(ctx, addr, uToken.Denom).Sub(uToken).Amount.Int64()
 	m.setCollateral(addr, uToken.Denom, newCollateral)
-	m.donatedCollateral = m.donatedCollateral.Add(uToken)
+	*m.donatedCollateral = m.donatedCollateral.Add(uToken)
```

Comment:
Fixes a bug in the mock keeper where `k.msgSrv.leveragekeeper` and `k.lk` were storing independent `donatedCollateral`, so one would update and the other not depending on how `DonateCollateral` was called.